pkg/cmd/token.go

```diff
@@ -0,0 +1,152 @@
+package cmd
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/puppetlabs/relay/pkg/client/openapi"
+	"github.com/puppetlabs/relay/pkg/config"
+	"github.com/puppetlabs/relay/pkg/debug"
+	"github.com/spf13/cobra"
+)
+
+func newTokensCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tokens",
+		Short: "Manage API tokens",
+		Args:  cobra.MinimumNArgs(1),
+	}
+
+	cmd.AddCommand(newCreateToken())
+	cmd.AddCommand(newListTokens())
+	cmd.AddCommand(newRevokeToken())
+
+	return cmd
+}
+
+func newCreateToken() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [token name]",
+		Short: "Create API token",
+		Args:  cobra.MinimumNArgs(1),
+		RunE:  doCreateToken,
+	}
+
+	cmd.Flags().BoolP("use", "u", true, "Configure the CLI to use the generated API token")
+	cmd.Flags().StringP("file", "f", "", "Write the generated token to the supplied file")
+
+	return cmd
+}
+
+func newRevokeToken() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "revoke [token id]",
+		Short: "Revoke API token",
+		Args:  cobra.MinimumNArgs(1),
+		RunE:  doRevokeToken,
+	}
+
+	return cmd
+}
+
+func newListTokens() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List API tokens",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  doListTokens,
+	}
+
+	cmd.Flags().BoolP("all", "a", false, "Show all account tokens")
+
+	return cmd
+}
+
+func doCreateToken(cmd *cobra.Command, args []string) error {
+	Dialog.Progress("Creating token...")
+
+	req := Client.Api.TokensApi.CreateToken(cmd.Context())
+	t, resp, err := Client.Api.TokensApi.CreateTokenExecute(
+		req.TokenRequest(openapi.TokenRequest{
+			Name: args[0],
+			Type: "user",
+		}),
+	)
+
+	if err != nil {
+		switch resp.StatusCode {
+		case http.StatusConflict:
+			return errors.New("A token by that name already exists")
+		default:
+			return err
+		}
+	}
+
+	if token, ok := t.GetTokenOk(); ok {
+		secret := token.UserTokenWithSecret.GetSecret()
+		filepath, err := cmd.Flags().GetString("file")
+		if err != nil {
+			return err
+		}
+
+		if filepath != "" {
+			if err := ioutil.WriteFile(filepath, []byte(secret), 0644); err != nil {
```

Comment:
Super minor thing that I also just thought of... you might want to open the file for writing prior to creating the token so that local file issues won't cause an unusable token to be created. Depending on complexity, might not be worth adding to this PR. Also, I wonder if a file mode that isn't world-writable might be better (0600?).